packages/examples/src/persistent-homology/persistent-homology.sty

```diff
@@ -142,21 +142,21 @@ forall Point p {
     center: p.circle1.center 
     width: p.circle1.r * 2.0
     height: p.circle1.r * 2.0
-    href: "shading.svg"
+    href: "persistent-homology-shading.svg"
```

Comment:
Thanks for reviewing this.  As you noted below, once an example is modified, we lose the example `root` information and no longer know which folder contains the image resource.  If there are different, for example, `shading.svg`s in different folders, it is ambiguous which one to choose.  The opportunity for name collisions in this case seems equivalent no matter if we access images via HTTP or local storage.

Reply:
Actually there's no need to cache image from previous sessions. Within a Style program, images should be unique, so we can cache them when we are in the example phase, and just retrieve the images in local mode.

Reply:
Thanks and a few clarifying questions since I am new to the new IDE.  How are we currently defining `session` and `workspace id`?  How is the `workspace id` accessed and when do we generate a new id?  In the described workflow, are we requiring a user to load an example with the stock images he or she wants prior to using them in their own diagram?  How do we want to invalidate the entries we create in local storage, particularly if they are workspace id specific -- whenever the `workspace id` changes?  Thanks for helping with the additional context!

Reply:
> Thanks and a few clarifying questions since I am new to the new IDE. How are we currently defining session and workspace id? How is the workspace id accessed and when do we generate a new id?

`WorkspaceMetadata` has the ID of the current workspace, which is roughly identified by a trio of Penrose programs. So whenever you load a new trio, `uuid()` is called to generate a new ID. You can simply search for all invocations for `uuid()` to see the call sites. The state management is done by recoil, so maybe look at https://recoiljs.org to see how it works.

> In the described workflow, are we requiring a user to load an example with the stock images he or she wants prior to using them in their own diagram?

Not sure if I understand the question. Image loading is done automatically if loading from gist, examples, or roger. Are you talking about the user uploading their own image? In that case, yeah I hope we can build that fairly easily (ie a simple file dialog can be a good starter). 

> How do we want to invalidate the entries we create in local storage, particularly if they are workspace id specific -- whenever the workspace id changes? 

Yeah, except when the workspaces are explicitly saved. You can get the list of these ids from `localFilesState`.

Reply:
Thanks, in the latest commit, local storage images are `workspace id` specific.  The editor was creating new `uuid`s when a workspace was saved, which caused saved workspaces to lose their images.  I removed the line of code creating the new uuid on workspace save.

It seems it would be better if the images were stored within the Recoil workspace state rather than outside of it.  For instance, when a saved workspace is deleted by the user, Recoil would also delete the images from local storage.  Right now these images accumulate each time a workspace with images is created and never go away. 

My suggestion is to move ahead with this PR since it fixes a bug in production and create an issue to move image resource management into Recoil state.